src/MainRouter.tsx

```diff
@@ -12,25 +13,93 @@ import DevconLounge from "./pages/DevconLounge/DevconLounge";
 import Profile from "./pages/Profile/Profile";
 import Gamification from "./components/Gamification/Gamification";
 import Agenda from "./pages/Agenda/Agenda";
-// import Categories from "./pages/Categories/Categories";
-
-export enum ROUTES {
-  APP = "/",
-  WELCOME1 = "/welcome1",
-  WELCOME2 = "/welcome2",
-  WELCOME3 = "/welcome3",
-  WELCOME4 = "/welcome4",
-  PROFILECREATION = "/profilecreation",
-  HOME = "/home",
-  RECENT = "/recent",
-  DEVCONLOUNGE = "/devconlounge",
-  PROFILE = "/profile",
-  GAMIFICATION = "/gamification",
-  AGENDA = "/agenda",
-  CATEGORIES = "/categories",
-}
+import {
+  ROUTES,
+  FIVE_MINNUTES,
+  FEEDTYPE_SEQUENCE,
+  ADDRESS_HEX_LENGTH,
+} from "./utils/constants";
+import { Session } from "./types/session";
 
 const MainRouter = (): ReactElement => {
+  const [sessions, setSessions] = useState<Session[]>([]);
+  const [sessionsReference, setSessionsReference] = useState<string>("");
+  const [isBeeRunning, setBeeRunning] = useState(false);
+
+  const bee = new Bee(process.env.BEE_API_URL || "http://localhost:1633/");
+  const rawFeedTopicSession = "sessions";
+
+  async function checkBee() {
+    fetch(process.env.BEE_API_URL + "addresses")
+      .then(async () => {
+        if (!isBeeRunning) {
+          setBeeRunning(true);
+          console.log("Bee is running");
+        }
+      })
+      .catch(() => {
+        setBeeRunning(false);
+        console.log("Bee stopped running");
+      });
+  }
+
+  async function getFeedUpdate() {
+    if (isBeeRunning) {
+      const sessionFeedTopic = bee.makeFeedTopic(rawFeedTopicSession);
+      const feedReader = bee.makeFeedReader(
+        FEEDTYPE_SEQUENCE,
+        sessionFeedTopic,
+        process.env.FEED_OWNER_ADDRESS as string
+      );
+      try {
+        const feedUpdateRes = await feedReader.download();
+        const feedRef = feedUpdateRes.reference as string;
+        if (feedRef !== sessionsReference) {
+          setSessionsReference(() => feedRef);
+          console.log("sessions reference updated: ", feedRef);
+        }
+      } catch (e) {
+        console.log("feed download error", e);
+      }
+    }
+  }
```

Comment:
Consider extracting all swarm related functionality to a separate file

Reply:
moved to bee.ts